### PR TITLE
Improved RFS readmes, added --help to RFS commands, fixed bug

### PR DESCRIPTION
**Pull request by chelma**
### Description
* Updated the READMEs for CreateSnapshot, MetadataMigration, DocumentsFromSnapshotMigration, and the main RFS library
* Updated the three RFS entrypoint tools to accept the `--help` option so they display their arguments
* Fixed a bug in the MetadataMigration entrypoint's CLI Argument parsing that was breaking

### Issues Resolved
* https://opensearch.atlassian.net/browse/MIGRATIONS-1840

### Testing
Ran the `--help` options for the CreateSnapshot, MetadataMigration, and DocumentsFromSnapshotMigration entrypoints; did a full local migration using the Docker Compose setup, S3, and the three entrypoints

### Check List
- [ ] New functionality includes testing
  - [ ] All tests pass, including unit test, integration test and doctest
- [X] New functionality has been documented
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by sumobrian — CHANGES_REQUESTED** [review]
Partially reviewed but discussed with @chelma today that we should add user guide document for RFS in the wiki. Will continue to review this one after those wiki changes are in.

File: CreateSnapshot/README.md [hunk 1]
```diff
@@ -4,14 +4,27 @@
...
+
+You can kick off the locally tool using Gradle.
+
+### S3 Snapshot
+
+From the root directory of the repo, run a CLI command like so:
 
-You can kick off the tool using Gradle.
+```
+./gradlew CreateSnapshot:run --args='--snapshot-name reindex-from-snapshot --s3-repo-uri s3://your-s3-uri --s3-region us-fake-1 --source-host http://hostname:9200'
```
[the remark this answers is not in the document]
Comment (chelma) [reply CreateSnapshot/README.md:21]: Do you have a recommendation how?  I seems like I need to embed HTML/Javascript to do this, and I imagine there's all sorts of problems with that.

[the remark this answers is not in the document]
Comment (sumobrian) [reply CreateSnapshot/README.md:21]: Prepend 'shell' just after your first set of backticks

**Comment by sumobrian** [comment]
> Partially reviewed but discussed with @chelma today that we should add user guide document for RFS in the wiki. Will continue to review this one after those wiki changes are in.

To add a little more clarity, we need additional documentation that is written for a user rather than a developer or someone who is more concerned with how the tooling works. Instead of addressing this as part of this PR, @chelma and I discussed put it in the wiki user guide which will eventually make its way to more formal documentation.

File: CreateSnapshot/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (sumobrian) [reply CreateSnapshot/README.md:21]: Nvm - just doesn't show up in preview mode.

**Comment by chelma** [comment]
Closed in favor of https://github.com/opensearch-project/opensearch-migrations/pull/832 to resolve a git branch conflict issue

**Pull request closed without merging**